### PR TITLE
Version 2.17

### PR DESCRIPTION
**Pull request by larskuhtz**
* [x] update cabal package version for 2.17
* [x] set service date for 2.17
* [x] finish CHANGELOG.md
* [x] update freeze file (no dependency changes, see below)

### CODE REVIEW

**Review by jwiegley — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+
+*   Remove error messages from pact output for on-chain transactions (#1543)
+*   Implement a per-tx timeout during creation of new blocks prevent mining
+    nodes from staling when block creation takes too long. (#1546)
+*   Miscellaneous changes for fine tuning of the gas model. (#1554)
+
+Bug fixes:
+
+*   Filter Module Cache for Coin. (#1548)
+*   Prevent table name clashes in module. (TODO)
```
Comment (larskuhtz) [inline CHANGELOG.md:24]:
```suggestion
*   Prevent table name clashes in module. (#1556)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+before that date.
+
+This version will expire on **2023-03-02T00:00:00Z**.
+
+To upgrade, pull the latest docker image or download the binary and restart the
+node.
+
+[Changes](https://github.com/kadena-io/chainweb-node/compare/2.16.1...2.17):
+
+*   Remove error messages from pact output for on-chain transactions (#1543)
```
Comment (larskuhtz) [inline CHANGELOG.md:16]:
```suggestion
*   Remove error messages from pact output for on-chain transactions. (#1543)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
 # `chainweb-node` Changelog
 
+## 2.17 (2022-11-15)
```
Comment (larskuhtz) [inline CHANGELOG.md:3]:
```suggestion
## 2.17 (2022-11-17)
```

**Comment by larskuhtz** [comment]
Dependency updates since 2.16.1:

```diff
209c209
<              pact -build-tool +cryptonite-ed25519 +no-advice -with-integer-gmp,
---
>              pact -build-tool +cryptonite-ed25519 +no-advice -tests-in-lib -with-integer-gmp,
362c362
< index-state: hackage.haskell.org 2022-11-08T03:14:15Z
---
> index-state: hackage.haskell.org 2022-11-11T17:41:49Z
```

File: cabal.project.freeze [hunk 4]
```diff
@@ -206,7 +206,7 @@ constraints: any.Boolean ==0.2.4,
              any.optparse-applicative ==0.17.0.0,
              optparse-applicative +process,
              any.pact ==4.4,
-             pact -build-tool +cryptonite-ed25519 +no-advice -with-integer-gmp,
+             pact -build-tool +cryptonite-ed25519 +no-advice -tests-in-lib -with-integer-gmp,
```
Comment (sirlensalot) [inline cabal.project.freeze:209]: why was this needed? it's off by default

File: src/Chainweb/Version.hs [hunk 5]
```diff
@@ -1017,8 +1017,8 @@ chainweb217Pact aoa v h = case aoa of
     At -> go (==) v h
     After -> go (<) v h
   where
-    go f Mainnet01 = f 3_250_357 -- 2022-12-02 00:00:00+00:00
-    go f Testnet04 = f 2_777_376 -- 2022-12-01 12:00:00+00:00
+    go f Mainnet01 = f 3_250_348 -- 2022-12-02 00:00:00+00:00
```
Comment (sirlensalot) [inline src/Chainweb/Version.hs:1020]: verified (I get 2022-12-01T23:59:17)

File: src/Chainweb/Version.hs [hunk 6]
```diff
@@ -1017,8 +1017,8 @@ chainweb217Pact aoa v h = case aoa of
...
+    go f Testnet04 = f 2_777_367 -- 2022-12-01 12:00:00+00:00
```
Comment (sirlensalot) [inline src/Chainweb/Version.hs:1021]: verified (I get 2022-12-01T12:00:26)

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,28 @@
...
+before that date.
+
+This version will expire on **2023-03-02T00:00:00Z**.
+
+To upgrade, pull the latest docker image or download the binary and restart the
+node.
+
+[Changes](https://github.com/kadena-io/chainweb-node/compare/2.16.1...2.17):
+
+*   Remove error messages from pact output for on-chain transactions. (#1543)
```
Comment (sirlensalot) [inline CHANGELOG.md:16]: Let's add a comment for builders: "Dapps can still retrieve transaction error messages from `local` endpoint."

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,28 @@
...
+*   Implement a per-tx timeout during creation of new blocks prevent mining
+    nodes from staling when block creation takes too long. (#1546)
```
Comment (sirlensalot) [inline CHANGELOG.md:18]:
```suggestion
    nodes from stalling when block creation takes too long. (#1546)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,28 @@
...
+*   Miscellaneous changes for fine tuning of the gas model. (#1554)
+
+Bug fixes:
+
+*   Filter Module Cache for Coin. (#1548)
```
Comment (sirlensalot) [inline CHANGELOG.md:23]:
```suggestion
*   Filter Module Cache for just `coin` contract. (#1548)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,28 @@
...
+
+This version will expire on **2023-03-02T00:00:00Z**.
+
+To upgrade, pull the latest docker image or download the binary and restart the
+node.
+
+[Changes](https://github.com/kadena-io/chainweb-node/compare/2.16.1...2.17):
+
+*   Remove error messages from pact output for on-chain transactions. (#1543)
+*   Implement a per-tx timeout during creation of new blocks prevent mining
```
Comment (sirlensalot) [inline CHANGELOG.md:17]:
```suggestion
*   Implement a per-tx timeout during creation of new blocks to prevent mining
```

File: cabal.project.freeze (hunk 4 above)
Comment (larskuhtz) [reply cabal.project.freeze:209]: This file is generate by `cabal freeze`, which adds it by default, even if default it off.

File: CHANGELOG.md (hunk 7 above)
Comment (larskuhtz) [inline CHANGELOG.md:16]:
```suggestion
*   Remove error messages from pact output for on-chain transactions. Dapps can 
    still retrieve transaction error messages from the `local` endpoint. (#1543)
```

**Review by sirlensalot — APPROVED** [review]
